ceph/ceph_admin/service_ops.py

```diff
@@ -0,0 +1,154 @@
+"""Module that interfaces with ceph orch  CLI."""
+from typing import Dict, Optional
+
+from .common import config_dict_to_string
+from .typing_ import OrchProtocol
+
+
+class ServiceOpsMixin:
+    """CLI for service operations using orch commands."""
+
```

Comment:
All the methods in this object have similar code with exception to the op i.e. `start | restart | reconfigure | redeploy`... would it be possible to create an internal method and pass the op as a parameter to it. Call this internal method varying the parameter. For example

```
def _service_op(cls, op, config):
  base_cmd = ["ceph", "orch"]
  if config.get("base_cmd_args"):
    ...
  base_cmd.append(op)
  base_cmd.append(config['service'])
  return cls.shell(args=base_cmd)

class ServiceOpMixin:

  def restart(...):
    return _service_op(self, 'restart', config)
```
This gives us the flexibility of overloading individual operations while having re-usability in place.

Reply:
Updated the changes. Logs attached - http://magna002.ceph.redhat.com/ceph-qe-logs/Preethi/cephci-run-1625744259988/

Reply:
Updated the changes. Logs attached - http://magna002.ceph.redhat.com/ceph-qe-logs/Preethi/cephci-run-1625744259988/